packages/alchemy/src/react/hooks/useAuthenticate.ts

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { useMutation } from "@tanstack/react-query";
+import { ClientOnlyPropertyError } from "../../config/errors.js";
+import type { AuthParams } from "../../signer/signer.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSigner } from "./useSigner.js";
+
+export function useAuthenticate() {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: authenticate,
+    isPending,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async (authParams: AuthParams) => {
+        if (!signer) {
+          throw new ClientOnlyPropertyError("signer");
+        }
+
+        const user = await signer.authenticate(authParams);
+
+        return user;
+      },
+    },
+    queryClient
```

Comment:
I had to do this because for w/e reason react-query exports hooks under `modern` or under `legacy`. `aa-alchemy` was using legacy and the demo app was using modern. doing it this way ensures a query client is always present